src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateCopy = { ...state };
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        for (const key in obj.extraData) {
+          stateCopy[key] = obj.extraData[key];
+        }
+        stateHistory.push({ ...stateCopy });
+        break;
+
+      case 'removeProperties':
+        for (const key of obj.keysToRemove) {
+          delete stateCopy[key];
+        }
+        stateHistory.push({ ...stateCopy });
+        break;
+
+      case 'clear':
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+        stateHistory.push({ ...stateCopy });
+        break;
+    }
```

Comment:
A `default` case is missing in the `switch` statement. It's important to have a `default` case for handling unexpected `type` values and possibly throwing an error or logging a warning.

Reply:
fixed